### PR TITLE
[data, model] feat: pad_to_length stability + Qwen3.5 ViT host-sync removal

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

Two independent changes:

1. **Data**: register `mm_token_type_ids` with `MainCollator` so the transformers-v5
   VLM token-type field packs / pads / SP-slices alongside `input_ids` instead of
   falling through to the ragged-list `default_collate` path.
2. **Model**: strip per-step host-device syncs from the Qwen3.5 / Qwen3.5-MoE
   vision tower. These ViT-side optimizations are ported from the existing
   `qwen3_vl` / `qwen3_vl_moe` patches.

Plus patchgen hygiene and a `flash-qla` pin bump.

### Changes

**1. `MainCollator` — register `mm_token_type_ids`**
- Add `"mm_token_type_ids": DataCollateInfo(-1, True, 0, 1)` to
  `DEFAULT_DATA_COLLATE_INFO` — the same entry `input_ids` uses, so it gets
  identical treatment: packed on the last dim, constant-0 padded under
  `pad_to_length`, and SP-padded(0) + SP-sliced. The pad region gets type-id 0
  (text), consistent with `input_ids`'s 0-pad and `image_mask` / `video_mask`'s
  `False`-pad. Text-only batches are unaffected (the key is absent → no-op).
  The field is produced by `data_transform` on the v5 path and consumed by the
  Qwen3.5 / Qwen3.5-MoE model forwards.

**2. Qwen3.5 ViT host-sync removal (ported from `qwen3_vl` / `qwen3_vl_moe`)**

These patches already exist in `veomni/models/transformers/qwen3_vl/` (see
`qwen3_vl_{gpu,npu}_patch_gen_config.py`); the Qwen3.5 ViT block is shared in
shape and the same syncs were hot in our profiles, so we mirror them into the
Qwen3.5 / Qwen3.5-MoE configs:

- `fast_pos_embed_interpolate`: vectorize the bilinear interpolation and drop
  the per-call `.tolist()` sync.
- `forward` Patch.5: pre-compute `vision_max_seqlen` once on the host and thread
  it via `kwargs` to each ViT block, instead of letting every block recompute
  `(cu_seqlens[1:] - cu_seqlens[:-1]).max()` and `.item()` it again inside
  `flash_attn_varlen_func`.
- The int hand-off is gated by `_VEOMNI_VISION_ATTENTION_PATCHED` (`True` on GPU
  / GPU-MoE where our `Qwen3_5VisionAttention.forward` override consumes it;
  `False` on NPU, which reuses upstream attention forward) **and**
  `is_flash_attention_requested(self.config)` (only FA's varlen consumes the int;
  eager / sdpa pop-and-discard, so it never leaks into `attention_interface(**kwargs)`).

**3. Patchgen hygiene**
- Drop `@auto_docstring` on `Qwen3_5CausalLMOutputWithLogProbs` and the MoE twin.
  HF's docstring resolver couldn't reach inherited field docs in the
  patchgen-generated parent and printed 9 `[ERROR]` lines on every module import.
  Plain docstring matches the canonical sibling in `veomni/utils/model_outputs.py`.

**4. Dependency**
- Bump `flash-qla` to 05/08/2026 main HEAD (PR #13). Picks up: backward returning
  the wrong number of gradients (#10), `auto_cp` UnboundLocalError (#8), `l2norm`
  dtype preservation (#13).

### Test

- `tests/data/test_collators.py`: new `test_vlm_collator_pad_to_length_sp_disabled`
  — VLM seq-len-keyed fields under `pad_to_length` (`mm_token_type_ids` + 3D mrope
  `position_ids` + `image_mask` / `video_mask`). Existing pad_to_length cases
  unchanged. **All pass.**
- `make quality` clean.
- `python -m veomni.patchgen.check_patchgen` clean — no `[ERROR]` lines, all 16
  configs up to date.

### API and Usage Example

N/A — purely additive. The `mm_token_type_ids` collate-info only acts on keys
present in the batch, so text-only and pre-v5 paths are untouched. The ViT
changes are internal to the Qwen3.5 / Qwen3.5-MoE generated modeling files.

### Touched areas

- `veomni/data/data_collator.py` — `mm_token_type_ids` collate-info entry.
- `veomni/models/transformers/qwen3_5{,_moe}/` patch-gen configs (GPU + NPU) and
  the regenerated `generated/` modeling `.py` / `.diff` — ViT host-sync removal +
  `@auto_docstring` drop, ported from `qwen3_vl` / `qwen3_vl_moe`.
- `pyproject.toml` / `uv.lock` — `flash-qla` pin bump.
- `tests/data/test_collators.py` — new VLM collator test case.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [x] Applied pre-commit checks
- [x] Added/updated documentation (registry entry self-documenting; comments inline)
- [x] Added tests to CI workflow

### CODE REVIEW

**Comment by TimYangst** [comment]
@gemini-code-assist please help to take a code review

File: veomni/models/transformers/qwen3_5/qwen3_5_gpu_patch_gen_config.py [hunk 1]
```diff
@@ -790,6 +816,89 @@ def qwen3_5_vision_model_dummy_forward(self):
...
+    )
+
+    if is_flash_attention_requested(self.config):
+        # Modification: prefer the int max_seqlen pre-computed once in
+        # Qwen3_5VisionModel.forward (Patch.5). Fall back to the original
+        # GPU-side reduction so this method still works when the model forward
+        # has not been patched (e.g. external callers, unit tests).
+        max_seqlen = kwargs.pop("vision_max_seqlen", None)
+        if max_seqlen is None:
+            max_seqlen = (cu_seqlens[1:] - cu_seqlens[:-1]).max()
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply veomni/models/transformers/qwen3_5/qwen3_5_gpu_patch_gen_config.py:880]: No an issue

File: veomni/models/transformers/qwen3_5/qwen3_5_gpu_patch_gen_config.py [hunk 2]
```diff
@@ -741,6 +754,19 @@ def qwen3_5_vision_model_forward(self, hidden_states: torch.Tensor, grid_thw: to
...
+    # `flash_attn_varlen_func` expects `max_seqlen_q/k` as Python ints; passing
+    # a 0-D GPU tensor forces an `.item()` inside the C++ binding. The HF body
+    # of Qwen3_5VisionAttention.forward recomputes `(cu_seqlens[1:] - cu_seqlens[:-1]).max()`
+    # per block, costing one host-device sync per ViT block per micro-batch
+    # (~32 blocks × micro_batches per step). We hoist the computation here so
+    # it happens once per ViT forward and thread the resulting int through
+    # `**kwargs` to every block; the patched Qwen3_5VisionAttention.forward
+    # picks it up via `vision_max_seqlen` and falls back to the original
+    # recompute when the key is absent (so non-VeOmni callers keep working).
+    kwargs["vision_max_seqlen"] = (cu_seqlens[1:] - cu_seqlens[:-1]).max().detach().cpu().item()
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply veomni/models/transformers/qwen3_5/qwen3_5_gpu_patch_gen_config.py]: Not an issue

**Comment by TimYangst** [comment]
@gemini-code-assist please help to take a code review

File: veomni/data/data_collator.py [hunk 3]
```diff
@@ -105,6 +105,7 @@ def __post_init__(self):
     "pixel_values_videos": DataCollateInfo(0, True, 0, 4),
     "image_mask": DataCollateInfo(-1, False, 0, 1),
     "video_mask": DataCollateInfo(-1, False, 0, 1),
+    "mm_token_type_ids": DataCollateInfo(-1, True, 0, 1),
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply veomni/data/data_collator.py:108]: Not an issue
